Add a convenient Bash script for installation (#1243)
open-telemetry/opentelemetry-dotnet-instrumentation: docs/README.md

Download and extract the appropriate binaries from
[the latest release](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases/latest).

> The path where you put the binaries is referenced as `$INSTALL_DIR`

> You can use a convenient Bash script to do it.
> Parameters:

Reviewer: ```suggestion
You can also use the `download.sh` script with the following parameters:
```

- `INSTALL_DIR` - location where binaries are to be installed;
  default: `./otel-dotnet-auto`
- `FLAVOUR` - possible values: `linux-glibc`, `linux-musl`, `macos`, `windows`;
  required
- `RELEASES_URL` - GitHub releases URL;
  default: `https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases`
- `TMPDIR` - temporary directory used when downloading the files;
  default: `$(mktemp -d)`  

> - `VERSION` - version to download;

Reviewer 2: nit: I guess that it should only be used for current version and above (just because it won't work for musl for the old ones and I'm not sure if it was tested against the old ones).

Reviewer: I'd turn this list into a table with columns `Parameter`, `Required`, and `Description`

  required

```sh
( set -o pipefail
curl -sSfL https://raw.githubusercontent.com/open-telemetry/opentelemetry-dotnet-instrumentation/main/download.sh |

>   VERSION=v0.3.1-beta.1 FLAVOUR=linux-glibc bash -s )

Reviewer 3: Extend release process to update this line.

Author: PTAL 83e999a3b

> The path where you put the binaries is referenced as `$INSTALL_DIR`

You can also use the [download.sh](../download.sh) script with the following parameters:

> | Parameter | Description | Required | Default value |

Reviewer 3: nit: https://marketplace.visualstudio.com/items?itemName=darkriszty.markdown-table-prettify :)

COR_PROFILER={918728DD-259F-4A6A-AC2B-B85E1B658318}
CORECLR_ENABLE_PROFILING=1
CORECLR_PROFILER={918728DD-259F-4A6A-AC2B-B85E1B658318}
DOTNET_ADDITIONAL_DEPS=$INSTALL_DIR/AdditionalDeps
DOTNET_SHARED_STORE=$INSTALL_DIR/store
DOTNET_STARTUP_HOOKS=$INSTALL_DIR/netcoreapp3.1/OpenTelemetry.AutoInstrumentation.StartupHook.dll
OTEL_DOTNET_AUTO_HOME=$INSTALL_DIR
OTEL_DOTNET_AUTO_INTEGRATIONS_FILE=$INSTALL_DIR/integrations.json
```

On **Windows** you need to additionally set:

```env

> COR_PROFILER_PATH_64=$INSTALL_DIR/win-x64/OpenTelemetry.AutoInstrumentation.Native.dll

Reviewer 2: On Windows this is not the correct way to expand an env var.

Author: What do you mean by "in Windows"? E.g. in PowerShell it is the correct way 😉
